src/Cli/dotnet/Commands/Pack/PackCommand.cs

```diff
@@ -40,16 +42,68 @@ public static PackCommand FromParseResult(ParseResult parseResult, string? msbui
             CommonOptions.PropertiesOption,
             CommonOptions.RestorePropertiesOption,
             PackCommandParser.TargetOption);
+
         return new PackCommand(
             parsedMSBuildArgs.CloneWithAdditionalProperties(projectLocator.GetCustomDefaultConfigurationValueIfSpecified()),
             noRestore,
             msbuildPath);
     }
 
+    public static int RunPackCommand(ParseResult parseResult)
+    {
+        var args = parseResult.GetValue(PackCommandParser.SlnOrProjectArgument)?.ToList() ?? new List<string>();
+
+        if (args.Count != 1)
+        {
+            Console.Error.WriteLine("Error: Only one .nuspec file can be packed at at time");
+            return 1;
+        }
+
+        var nuspecPath = args[0];
+
+        var packArgs = new PackArgs()
+        {
+            Path = nuspecPath,
+            BasePath = Path.GetDirectoryName(nuspecPath),
+            Logger = new NuGetConsoleLogger(),
```

Comment:
I would _really_ like to not use the default NuGetConsoleLogger here - we have feedback from users that it is not in line with the rest of the SDK output. Please strongly consider making a logger interface implementation that uses our Reporter concepts and verbosity to render a more pleasant view to the user.

Reply:
Thanks for pointing this out - the thing I'm concerned about is spamming the console output here. It looks like the `PackArgs` has a LogLevel property - can you map the current CLI VerbosityOptions to the NuGet LogLevel concept and flow that through as well so that the output is responsive to what the user requests?